hibernate-core/src/main/java/org/hibernate/boot/model/source/internal/hbm/ModelBinder.java

```diff
@@ -281,6 +282,20 @@ private void bindRootEntity(EntityHierarchySourceImpl hierarchySource, RootClass
 				rootEntityDescriptor
 		);
 
+		if (database.getDialect().isNonNullPrimaryKeyRequired()) {
```

Comment:
I wonder if a better option is a "complete()" method on Database where we can then iterate all tables and do this for all pks